### ISSUE TITLE
publish each image to its own topic for use in rqt_image or rviz2

### ISSUE THREAD

**Issue by flynneva** [MatthiasKillat/ros2_shm_vision_demo#11]
- each processing executable should probably have its own topic to publish to
- this would enable users to use rqt and rviz2 as well
- additional copies would be avoided using iceoryx I think, making the performance gain for the `shm_demo.sh` even more apparent

**Comment by ZhenshengLee** [comment]
@flynneva Hi, see you again!

> * each processing executable should probably have its own topic to publish to

What do you mean about its own topic? rviz2 and rqt_image can only accept the standard sensor_images::msg::Image, rather than  [ros2_shm_vision_demo](https://github.com/MatthiasKillat/ros2_shm_vision_demo)/[msg](https://github.com/MatthiasKillat/ros2_shm_vision_demo/tree/main/msg)/Image.msg

So, there is an idea of developing tools of showing custom defined image msg, which is ugly, but will be useful for debugging.

What do you think?

There will be some PRs about custom image definition conversion from/to opencv image in https://github.com/ZhenshengLee/ros2_shm_msgs , would you like to give it a little reviews?

An invitation has been sent to you though.

**Comment by flynneva** [comment]
> What do you mean about its own topic?

@ZhenshengLee I mean that currently the demo launches its own GUI (non-rqt) window to visualize the images. These images are not published to a topic but rather just shown in the GUI.

We should publish these images to a topic (sensor_msgs::msg::Image most likely) so that we can use normal ROS visualization tools (rqt, rviz, foxglove, etc.)

**Comment by ZhenshengLee** [comment]
> We should publish these images to a topic (sensor_msgs::msg::Image most likely) so that we can use normal ROS visualization tools (rqt, rviz, foxglove, etc.)

Agreed, check the readme of my repo if interested https://github.com/ZhenshengLee/ros2_shm_msgs/blob/master/README.md

There is a component shm_image_bridge for visualization of a shm_image_topic.
